From e79e5362bb90b20d868ac1b593a23553beb638a1 Mon Sep 17 00:00:00 2001
From: mazbrili <mazbrili@yahoo.com>
Date: Sat, 20 Oct 2018 18:56:36 +0700
Subject: apply path difference

change path with builtin QStandardPaths from Qt

modify pro files to use pkgconfig

diff --git a/bookinfo.cpp b/bookinfo.cpp
old mode 100644
new mode 100755
index f75bcaa..35a7d3e
--- a/bookinfo.cpp
+++ b/bookinfo.cpp
@@ -19,26 +19,29 @@
  * Boston, MA 02110-1301, USA.
  */
 #include "bookinfo.h"
-#if QT_VERSION >= 0x050000
-#include <poppler-qt5.h>
-#else
-#include <poppler-qt4.h>
-#endif
+
+
 
 #include<QApplication>
 #include<QInputDialog>
 #include<QDir>
 #include<QFileInfo>
 #include<QSettings>
+#include<QStandardPaths>
 //#define thumbnails QDir::homePath()+"/."+QApplication::applicationName()+"/thumbnails"
 
 BookInfo::BookInfo()
 {
 
 QSettings settings;
-
+#if not defined(Q_OS_HAIKU)
 QString appdir=(settings.value("BooksPath",
                                              QDir::homePath()+"/."+QApplication::applicationName()).toString());
+#elif defined(Q_OS_HAIKU)
+QString appdir=(settings.value("BooksPath",
+											QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)).toString());
+											
+#endif
 mThumbnails=appdir+"/thumbnails/";
 
 }
diff --git a/bookinfo.h b/bookinfo.h
old mode 100644
new mode 100755
index 445a9a7..0dc6f75
--- a/bookinfo.h
+++ b/bookinfo.h
@@ -23,7 +23,7 @@
 #define BOOKINFO_H
 #include <QString>
 #if QT_VERSION >= 0x050000
-#include <poppler-qt5.h>
+#include <poppler/qt5/poppler-qt5.h>
 #else
 #include <poppler-qt4.h>
 #endif
diff --git a/booksorganizer.pro b/booksorganizer.pro
old mode 100644
new mode 100755
index 2162d85..8764eac
--- a/booksorganizer.pro
+++ b/booksorganizer.pro
@@ -27,7 +27,13 @@ win32{
     LIBS         += -L poppler-win\bin -lpoppler-qt5
     RC_FILE = myapp.rc
 #    QT +=winextras
-}else{
+}
+haiku{
+	TARGET = BooksOrg
+	LIBS += -lpoppler -lpoppler-qt5
+	CONFIG += link_pkgconfig
+	PKGCONFIG += poppler-qt5
+	}else{
 
         equals(QT_MAJOR_VERSION, 5) {
           INCLUDEPATH  += /usr/include/poppler/qt5
diff --git a/configuredialog.cpp b/configuredialog.cpp
index 4058fa3..0cefaba 100644
--- a/configuredialog.cpp
+++ b/configuredialog.cpp
@@ -4,15 +4,23 @@
 #include <QSettings>
 #include <QMessageBox>
 #include <QFileDialog>
+#include <QStandardPaths>
+
 ConfigureDialog::ConfigureDialog(QWidget *parent) :
     QDialog(parent),
     ui(new Ui::ConfigureDialog)
 {
     ui->setupUi(this);
     QSettings settings;
-
+    #if not defined(Q_OS_HAIKU)
     ui->lineEditPath->setText(settings.value("BooksPath",
                                              QDir::homePath()+"/."+QApplication::applicationName()).toString());
+    #elif defined(Q_OS_HAIKU)
+    ui->lineEditPath->setText(settings.value("BooksPath",
+											QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)).toString());
+											
+#endif
+    
     ui->checkBoxCopy->setChecked(settings.value("CopyBooks",false).toBool());
 
 }
diff --git a/documentview.h b/documentview.h
old mode 100644
new mode 100755
index dd911c0..7e68d96
--- a/documentview.h
+++ b/documentview.h
@@ -33,7 +33,8 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
 
 #include <QtWidgets>
 #include <QtPrintSupport>
-#include <poppler-qt5.h>
+#include <poppler/qt5/poppler-qt5.h>
+
 
 #else
 #include <poppler-qt4.h>
diff --git a/main.cpp b/main.cpp
index e45955e..35a8bff 100644
--- a/main.cpp
+++ b/main.cpp
@@ -57,9 +57,14 @@ int main(int argc, char *argv[])
     qRegisterMetaType< QList< QRectF > >("QList<QRectF>");
 
     QApplication a(argc, argv);
+    
+#if defined(Q_OS_WIN32)
     a.setApplicationName("booksorg");
+#elseif defined(Q_OS_HAIKU)
+	a.setApplicationName("BooksOrg");
+#endif
     a.setOrganizationName("elokab");
-     a.setApplicationVersion("0.3");
+    a.setApplicationVersion("0.3");
     a.setWindowIcon(QIcon(":/icons/booksorg"));
 
 
@@ -93,12 +98,18 @@ int main(int argc, char *argv[])
     a.setApplicationDisplayName(QObject::tr("Books Organizer"));
 #endif
 
-#ifdef Q_OS_WIN32
+#if defined(Q_OS_WIN32)
     associateFileTypes(QStringList(".pdf"));
-#endif
-
     QDir dir;
     dir.mkpath(QDir::homePath()+"/."+a.applicationName()+"/thumbnails");
+#elif defined(Q_OS_HAIKU)
+       QDir dir(QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation));
+      // dir.mkdir(a.applicationName());
+   	 dir.mkpath(QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)+
+   	   QDir::separator()+"thumbnails");
+#endif
+
+  
 
     MainWindow w;
     w.show();
diff --git a/mainwindow.cpp b/mainwindow.cpp
index d1f46a0..4a2b55f 100644
--- a/mainwindow.cpp
+++ b/mainwindow.cpp
@@ -67,10 +67,13 @@ QSettings settings;
     //---------------------------------------------
 
 
-
+#if defined(Q_OS_HAIKU)
+   Path::setWorkingDir(settings.value("BooksPath",
+                          QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)).toString());
+#elif
    Path::setWorkingDir(settings.value("BooksPath",
                              QDir::homePath()+"/."+QApplication::applicationName()).toString());
-
+#endif
    Path::setCopyBooks(settings.value("CopyBooks",false).toBool());
 
 
diff --git a/pageitem.h b/pageitem.h
old mode 100644
new mode 100755
index 3606696..f969b8e
--- a/pageitem.h
+++ b/pageitem.h
@@ -26,8 +26,7 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
 #include <QtGui>
 
 #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <poppler-qt5.h>
-
+#include <poppler/qt5/poppler-qt5.h>
 #include <QtWidgets>
 #include <QtConcurrent>
 #else
diff --git a/path.cpp b/path.cpp
index 069e2be..ec35c18 100644
--- a/path.cpp
+++ b/path.cpp
@@ -1,3 +1,8 @@
 #include "path.h"
+#include <QStandardPaths>
+#if defined(Q_OS_HAIKU)
+QString Path::mWorkingDir=QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
+#elif
 QString Path::mWorkingDir= QDir::homePath()+"/."+QApplication::applicationName();
+#endif
 bool Path::mCopy=false;
diff --git a/searchthread.h b/searchthread.h
old mode 100644
new mode 100755
index 5e36df5..9028025
--- a/searchthread.h
+++ b/searchthread.h
@@ -31,7 +31,7 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
 #include <QStandardItemModel>
 
 #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <poppler-qt5.h>
+#include <poppler/qt5/poppler-qt5.h>
 #else
 #include <poppler-qt4.h>
 #endif // QT_VERSION
diff --git a/searchtools.cpp b/searchtools.cpp
old mode 100644
new mode 100755
index 3bf60ca..c64a4a4
--- a/searchtools.cpp
+++ b/searchtools.cpp
@@ -24,6 +24,7 @@
 #include "eicon.h"
 #include <QtGui>
 #include <QSettings>
+#include <QAction>
 
 SearchTools::SearchTools(bool isPdf, QWidget *parent) :
     QWidget(parent),
-- 
2.19.0

